tests/tasty/Arbitrary.hs

```diff
@@ -155,3 +174,109 @@ instance Arbitrary Constant where
                     , L <$> arbitrary <*> arbitrary
                     ]
   shrink = genericShrink
+
+-- | Used in UndoANFTests.
+newtype AnfSymbol = AnfSymbol { unAnfSymbol :: Symbol }
+  deriving (Eq, Show, Generic)
+instance Arbitrary AnfSymbol where
+  arbitrary = AnfSymbol . mappendSym anfPrefix <$> arbitrary
+
+-- | This instance does **not** create Refts with anf symbols.
+instance Arbitrary Reft where
+  arbitrary = reft <$> arbitrary <*> arbitrary
+
+-- | This instance does **not** create SortedRefts with anf symbols.
+instance Arbitrary SortedReft where
+  arbitrary = arbitrarySortedReft (const arbitrary) 1
+
+arbitrarySortedReft :: (Int -> Gen Symbol) -> Int -> Gen SortedReft
+arbitrarySortedReft symGen = \i -> do
+  sort <- arbitrary
+  eq <- oneof $ pure <$> [Eq, Ueq]
+  sym <- symGen i
+  expr <- arbitrary
+  pure $ RR sort $ reft sym (PAtom eq (EVar sym) expr)
+
+-- | Base environment with no declared properties; do not add an Arbitrary
+-- instance to this and instead use newtypes.
+newtype Env = Env { unEnv :: M.HashMap Symbol SortedReft }
+  deriving (Eq, Show, Generic)
+
+-- | Env without anf vars.
+newtype NoAnfEnv = NoAnfEnv { unNoAnfEnv :: Env }
+  deriving (Eq, Show, Generic)
+instance Arbitrary NoAnfEnv where
+  arbitrary = NoAnfEnv <$> (arbitraryEnv 4 gen)
+    where
+      -- | Note that this relies on the property that the Arbitrary instance for
+      -- Symbol cannot create lq_anf$ vars.
+      symGen = const arbitrary
+      reftGen = const arbitrary
+      gen = fmap pure . (pairGen symGen reftGen)
+
+pairGen :: (Int -> Gen a) -> (Int -> Gen b) -> Int -> Gen (a, b)
+pairGen aGen bGen = \i -> do
+  a <- aGen i
+  b <- bGen i
+  pure (a, b)
+
+-- | Env with anf vars that do not reference further anf vars.
+newtype FlatAnfEnv = FlatAnfEnv { unFlatAnfEnv :: Env }
+  deriving (Eq, Show, Generic)
+instance Arbitrary FlatAnfEnv where
+  arbitrary = FlatAnfEnv <$> (arbitraryEnv 4 gen)
+    where
+      symGen = const arbitrary
+      anfGen' = anfGen symGen (const arbitrary)
+      gen = finalAnfGen (fmap pure . anfGen')
+
+anfGen :: (Int -> Gen AnfSymbol) -> (Int -> Gen SortedReft) -> Int -> Gen (Symbol, SortedReft)
+anfGen symGen sreftGen = \i -> do
+  sym <- unAnfSymbol <$> symGen i
+  sreft <- sreftGen i
+  pure $ (sym, sreft)
```

Comment:
Why not
```suggestion
      anfGen = (\a b -> [(unAnfSymbol a, b)]) <$> arbitrary <*> arbitrary
      gen = finalAnfGen anfGen
```
?

Reply:
Changed structure to something different.  The anfGens are now inlined and can generate `Env`s of larger sizes.